src/import/sst_service.rs

```diff
@@ -374,40 +430,96 @@ where
+                    meta.get_is_delete(),
+                    meta.get_cf(),
+                    context.clone(),
+                );
+                match importer.do_apply_kv_file(
+                    meta.get_restore_ts(),
+                    temp_file,
+                    req.get_rewrite_rule(),
+                    &mut build_req_fn,
+                ) {
+                    Ok(range) => {
+                        drop(build_req_fn);
+                        if !reqs.is_empty() {
+                            let cmd = make_request(reqs.as_mut(), context);
+                            cmd_reqs.push(cmd);
+                        }
+                        for cmd in cmd_reqs {
+                            let (cb, future) = paired_future_callback();
+                            match router.send_command(
+                                cmd,
+                                Callback::write(cb),
+                                RaftCmdExtraOpts::default(),
+                            ) {
+                                Ok(_) => futs.push(future),
+                                Err(_e) => {
+                                    let mut import_err = kvproto::import_sstpb::Error::default();
+                                    import_err
+                                        .set_message("failed to send raft command".to_string());
+                                    apply_resp.set_error(import_err);
+                                }
+                            }
+                        }
+                        if let Some(r) = range {
+                            apply_resp.set_range(r);
+                        }
+                    }
+                    Err(e) => {
+                        apply_resp.set_error(e.into());
+                    }
+                }
+            }
+            Err(e) => {
+                apply_resp.set_error(e.into());
+            }
+        }
+
         let handle_task = async move {
             // Records how long the apply task waits to be scheduled.
             sst_importer::metrics::IMPORTER_APPLY_DURATION
                 .with_label_values(&["queue"])
                 .observe(start.saturating_elapsed().as_secs_f64());
-
-            let res = importer.apply::<E>(
-                req.get_meta(),
-                req.get_storage_backend(),
-                req.get_rewrite_rule(),
-                limiter,
-                engine,
-            );
-            let mut resp = ApplyResponse::default();
-            match res {
-                Ok(range) => {
-                    if let Some(r) = range {
-                        resp.set_range(r);
+            let resp = if !futs.is_empty() {
+                Ok(join_all(futs).await.iter().fold(apply_resp, |mut resp, x| {
+                    if x.is_err() {
+                        let mut import_err = kvproto::import_sstpb::Error::default();
+                        import_err.set_message("failed to complete raft command".to_string());
```

Comment:
Maybe collect those errors and send back to BR?

Reply:
fixed